### ISSUE TITLE
vgl doesn't work in vue-cli's simple webpack setting

### ISSUE THREAD

**Issue by soyvv** [eddow/vue-golden-layout#5]
Hi,
I've tried to experiment vgl in a simplistic setting generated by vue-cli.
here's what I have done:
`vue init webpack-simple vgl-test`
and made a few webpack config change (style loader). I copied the template and the code from another closed [issue](https://github.com/eddow/vue-golden-layout/issues/2), and somehow cannot make it work. 

As in the issue mentioned above, I also had a warning saying:

> WARNING in ./~/vue-golden-layout/dist/vue-golden-layout.js
> 4:25-32 Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
> 

In the browser console, I did see a lot of vue and vgl related error, which I have no idea whether is related with the warning above (or some stupid mistakes I made elsewhere): 

> ---> <LayoutGolden>
>        <App> at /home/zhangzhikai/dev/vgl-test/src/App.vue
>          <Root>
> warn @ webpack-internal:///17:480
> webpack-internal:///17:567 TypeError: _super.apply is not a function
>     at new goldenContainer (webpack-internal:///102:294)
>     at collectDataFromConstructor (webpack-internal:///100:52)
>     at VueComponent.data (webpack-internal:///100:107)
>     at VueComponent.mergedDataFn (webpack-internal:///17:1146)
>     at getData (webpack-internal:///17:3195)
>     at initData (webpack-internal:///17:3154)
>     at initState (webpack-internal:///17:3085)
>     at VueComponent.Vue._init (webpack-internal:///17:4225)
>     at new VueComponent (webpack-internal:///17:4397)
>     at createComponentInstanceForVnode (webpack-internal:///17:3679)
> handleError @ webpack-internal:///17:567
> webpack-internal:///17:480 [Vue warn]: Error in data(): "TypeError: _super.apply is not a function"

Could you help make this flow work (like out of box work) so that vue users can easily taste golden layout? Thanks!

Also, as a side note, I tried to run the code on its own (as the readme doc directs), npm run compileTest actually generate a static html file which seemed to work fine.

Thank you!

**Comment by eddow** [comment]
Hi,
Thank you for reporting the error.
I think I have an idea of the problem and I made a patch. Could you please try to `npm update` to have version `1.0.10` then tell me if the error is corrected?

**Comment by soyvv** [comment]
[localhost-1503633322518.log.zip](https://github.com/eddow/vue-golden-layout/files/1250785/localhost-1503633322518.log.zip)
Hi,
I've tried that -- the warning went away but the error was still there. I've packed all the console error log in the attachment. Hope this helps your debug:)

**Comment by eddow** [comment]
So, I see. May I pretty please ask you to create a repository with the non-compiling project? Even if it is destroyed afterward, just to be able to reproduce easily.
Thank you in advance,

**Comment by soyvv** [comment]
sorry for the delay!
I've created a repo here:
https://github.com/soyvv/vue4vgl-test

**Comment by eddow** [comment]
Oh, hi again.

In the mean time, there are some administrative stuff happening on fuse-box (2 month to discuss a PR) and I am not sure I will be able to fix anything there for now (I am the first one using the bundler to make an npm package)

Byt the time, I changed a bit the [installation procedure](https://github.com/eddow/vue-golden-layout#usage) ;-)

Basically, either use fuse-box to bundle your project (it works pretty well for regular projects) either bundle the sources. The only thing is that it is in vue/typescript and need to be compiled.

For the remaining, I of course try to see what has to be fixed (thanks for the reop) and all this annoyance to use the sources is quite temporary.

**Comment by eddow** [comment]
Okay, here is the diagnostic and the solution.
1- there was a reference missing in the library, this was solved and published (fuse-box internal config)
2- there was a conflict in the `vue` version used. I just added in the doc that the `vue` alias ought to be `'vue/dist/vue.common.js'` instead of `'vue/dist/vue.esm.js'`

To make the project work  :
- `npm update vue-golden-layout`
- in the `webpack.config.js`, change `esm` to `common` here :
```js

  resolve: {
    alias: {
      'vue$': 'vue/dist/vue.esm.js'
    }
  },
```
Notes :
- I couldn't find a clear documentation of the difference between `esm` and `common`. If needed, we could imagine 2 builts of `vgl` (it changes the way the `vue` object is exported) What I find most when googleing `vue esm` is people complaining it raises errors
- I am still unsure of the success of the CSS import, I keep on on this part
